lib/ui/platform_dispatcher.dart

```diff
@@ -0,0 +1,1586 @@
+/// See [PlatformDispatcher.instance] for more information about why this is
+/// preferred.
+class PlatformDispatcher {
+  /// Private constructor, since only dart:ui is supposed to create one of
+  /// these.
+  PlatformDispatcher._() {
+    _setNeedsReportTimings = _nativeSetNeedsReportTimings;
+  }
+
+  /// The [PlatformDispatcher] singleton.
+  ///
+  /// Consider avoiding static references to this singleton though
+  /// [PlatformDispatcher.instance] and instead prefer using a binding for
+  /// dependency resolution such as `WidgetsBinding.instance.platformDispatcher`.
+  ///
+  /// Static access of this object means that Flutter has few, if any options to
+  /// fake or mock the given object in tests. Even in cases where Dart offers
+  /// special language constructs to forcefully shadow such properties, those
+  /// mechanisms would only be reasonable for tests and they would not be
+  /// reasonable for a future of Flutter where we legitimately want to select an
+  /// appropriate implementation at runtime.
+  ///
+  /// The only place that `WidgetsBinding.instance.platformDispatcher` is
+  /// inappropriate is if access to these APIs is required before the binding is
+  /// initialized by invoking `runApp()` or
+  /// `WidgetsFlutterBinding.instance.ensureInitialized()`. In that case, it is
+  /// necessary (though unfortunate) to use the [PlatformDispatcher.instance]
+  /// object statically.
+  static PlatformDispatcher get instance => _instance;
+  static final PlatformDispatcher _instance = PlatformDispatcher._();
+
+  /// The current platform configuration.
+  ///
+  /// If values in this configuration change, [onPlatformConfigurationChanged]
+  /// will be called.
+  PlatformConfiguration get configuration => _configuration;
+  PlatformConfiguration _configuration = const PlatformConfiguration();
+
+  /// Called when the platform configuration changes.
+  ///
+  /// The engine invokes this callback in the same zone in which the callback
+  /// was set.
+  VoidCallback? get onPlatformConfigurationChanged => _onPlatformConfigurationChanged;
+  VoidCallback? _onPlatformConfigurationChanged;
+  Zone _onPlatformConfigurationChangedZone = Zone.root;
+  set onPlatformConfigurationChanged(VoidCallback? callback) {
+    _onPlatformConfigurationChanged = callback;
+    _onPlatformConfigurationChangedZone = Zone.current;
+  }
+
+  /// The current list of views, including top level platform windows used by
+  /// the application.
+  ///
+  /// If any of their configurations change, [onMetricsChanged] will be called.
+  Iterable<FlutterView> get views => _views.values;
+  Map<Object, FlutterView> _views = <Object, FlutterView>{};
+
+  // A map of opaque platform view identifiers to view configurations.
+  Map<Object, ViewConfiguration> _viewConfigurations = <Object, ViewConfiguration>{};
+
+  /// A callback that is invoked whenever the [ViewConfiguration] of any of the
```

Comment:
It appears that there's no easy way to figure out which view has changed. Is that gonna be a problem?